target/linux/ramips/image/common-sercomm.mk

```diff
@@ -0,0 +1,87 @@
+DEVICE_VARS += SERCOMM_KERNEL_OFFSET SERCOMM_ROOTFS_OFFSET
+
+define Build/sercomm-crypto
+	$(TOPDIR)/scripts/sercomm-crypto.py \
+		--input-file $@ \
+		--key-file $@.key \
+		--output-file $@.ser \
+		--version $(SERCOMM_SWVER)
+	$(STAGING_DIR_HOST)/bin/openssl enc -md md5 -aes-256-cbc \
+		-in $@ \
+		-out $@.enc \
+		-K `cat $@.key` \
+		-iv 00000000000000000000000000000000
+	dd if=$@.enc >> $@.ser 2>/dev/null
+	mv $@.ser $@
+	rm -f $@.enc $@.key
+endef
+
+define Build/sercomm-kernel
+	$(TOPDIR)/scripts/sercomm-kernel-header.py \
+		--kernel-image $@ \
+		--kernel-offset $(SERCOMM_KERNEL_OFFSET) \
+		--rootfs-offset $(SERCOMM_ROOTFS_OFFSET) \
+		--output-header $@.hdr
+	dd if=$@ >> $@.hdr 2>/dev/null
+	mv $@.hdr $@
+endef
+
+define Build/sercomm-part-tag
+	$(call Build/sercomm-part-tag-common,$(word 1,$(1)) $@)
+endef
+
+define Build/sercomm-part-tag-common
+	$(eval file=$(word 2,$(1)))
+	$(TOPDIR)/scripts/sercomm-partition-tag.py \
+		--input-file $(file) \
+		--output-file $(file).tmp \
+		--part-name $(word 1,$(1)) \
+		--part-version $(SERCOMM_SWVER)
+	mv $(file).tmp $(file)
+endef
+
+define Build/sercomm-payload
+	$(TOPDIR)/scripts/sercomm-pid.py \
+		--hw-version $(SERCOMM_HWVER) \
+		--hw-id $(SERCOMM_HWID) \
+		--sw-version $(SERCOMM_SWVER) \
+		--pid-file $@.pid \
+		--extra-padding-size 0x10 \
+		--extra-padding-first-byte 0x0a
+	$(TOPDIR)/scripts/sercomm-payload.py \
+		--input-file $@ \
+		--output-file $@.tmp \
+		--pid "$$(cat $@.pid | od -t x1 -An -v | tr -d '\n')"
+	mv $@.tmp $@
+	rm $@.pid
+endef
+
+define Build/sercomm-prepend-tagged-kernel
+	$(CP) $(IMAGE_KERNEL) $(IMAGE_KERNEL).tagged
+	$(call Build/sercomm-part-tag-common,$(word 1,$(1)) \
+		$(IMAGE_KERNEL).tagged)
+	dd if=$@ >> $(IMAGE_KERNEL).tagged
+	mv $(IMAGE_KERNEL).tagged $@
+endef
+
+define Device/sercomm_dxx
+  $(Device/dsa-migration)
+  BLOCKSIZE := 128k
+  PAGESIZE := 2048
+  KERNEL_SIZE := 6144k
+  UBINIZE_OPTS := -E 5
+  LOADER_TYPE := bin
+  KERNEL_LOADADDR := 0x81001000
+  LZMA_TEXT_START := 0x82800000
+  KERNEL := kernel-bin | append-dtb | lzma | loader-kernel | lzma | \
+	uImage lzma | sercomm-kernel
```

Comment:
The kernel is compressed twice. Is it possible to skip one of them? Could you try to drop the first lzma and replace loader-kernel with relocate-kernel, or drop the second lzma and use uImage none instead of uImage lzma for the uImage header?

Reply:
@981213 glad to see you here! Yeah, I tried it again. 
1st option (dropping the first lzma and replace loader-kernel with relocate-kernel) - it hangs on "Starting kernel":
```
## Booting image at bc400100 ...
   Image Name:   MIPS OpenWrt Linux-5.15.49
   Image Type:   MIPS Linux Kernel Image (lzma compressed)
   Data Size:    2907229 Bytes =  2.8 MB
   Load Address: 81001000
   Entry Point:  81001000
   Verifying Checksum ... OK
   Uncompressing Kernel Image ... OK
kernel addr :0xBC400100
No initrd
## Transferring control to Linux (at address 81001000) ...
## Giving linux memsize in MB, 256

Starting kernel ...
```
2nd option (dropping the second lzma and use uImage none instead of uImage lzma for the uImage header):
```
## Booting image at bc400100 ...
   Image Name:   MIPS OpenWrt Linux-5.15.49
   Image Type:   MIPS Linux Kernel Image (uncompressed)
   Data Size:    2868290 Bytes =  2.7 MB
   Load Address: 81001000
   Entry Point:  81001000
   Verifying Checksum ... OK
Unimplemented compression type 0
kernel check fail
```
We can use fast compression for the second lzma (-a0). What do you think? 

Successful boot (current recipe from the commit):
```
*************************************
Boot Flag : Sercomm0
*************************************
Fw header Magic check OK!
FW header checksum should: 0x7ec29420, crc32 result: 0x7ec29420, Fw header CRC c                                                                                                             heck OK!
kernel : real offset: 0x00400100, data length: 0x002c5c9d,checksum :0xe93a226b
kernel result: 0xe93a226b, kernel check sum ok!
rootfs:  real offset: 0x01000000, data length: 0x00000004, checksum :0x2d55fc1c
rootfs result: 0x2d55fc1c, rootfs check sum ok!
Kernel Addr : 0xbc400100
## Booting image at bc400100 ...
   Image Name:   MIPS OpenWrt Linux-5.15.49
   Image Type:   MIPS Linux Kernel Image (lzma compressed)
   Data Size:    2907229 Bytes =  2.8 MB
   Load Address: 81001000
   Entry Point:  81001000
   Verifying Checksum ... OK
   Uncompressing Kernel Image ... OK
kernel addr :0xBC400100
No initrd
## Transferring control to Linux (at address 81001000) ...
## Giving linux memsize in MB, 256

Starting kernel ...



OpenWrt kernel loader for MIPS based SoC
Copyright (C) 2011 Gabor Juhos <juhosg@openwrt.org>
Decompressing kernel... done!
Starting kernel at 80001000...
```

Reply:
> We can use fast compression for the second lzma (-a0). What do you think?

OK. Let's go with that for now.

Reply:
> > We can use fast compression for the second lzma (-a0). What do you think?
> 
> OK. Let's go with that.

Thanks, done. Tests are ok.